bindings/sundials4py/examples/arkode/ark_heat1D.py

```diff
@@ -0,0 +1,212 @@
+# centered differences, with the data distributed over N points
+# on a uniform spatial grid.
+#
+# This program solves the problem with either an ERK or DIRK
+# method.  For the DIRK method, we use a Newton iteration with
+# the SUNLinSol_PCG linear solver, and a user-supplied Jacobian-vector
+# product routine.
+#
+# 100 outputs are printed at equal intervals, and run statistics
+# are printed at the end.
+# -----------------------------------------------------------------
+
+import numpy as np
+from sundials4py.core import *
+from sundials4py.arkode import *
+
+
+class Heat1DProblem:
+    def __init__(self, N, k):
+        self.N = N
+        self.k = k
+        self.dx = 1.0 / (N - 1)
+
+    def set_init_cond(self, yvec):
+        y = N_VGetArrayPointer(yvec)
+        y[:] = 0.0
+        return 0
+
+    def f(self, t, yvec, ydotvec):
+        N, k, dx = self.N, self.k, self.dx
+        Y = N_VGetArrayPointer(yvec)
+        Ydot = N_VGetArrayPointer(ydotvec)
+        Ydot[:] = 0.0
+        c1 = k / dx / dx
+        c2 = -2.0 * k / dx / dx
+        # Vectorized Laplacian
+        Ydot[1:-1] = c1 * Y[:-2] + c2 * Y[1:-1] + c1 * Y[2:]
+        # Dirichlet BCs
+        Ydot[0] = 0.0
+        Ydot[-1] = 0.0
+        # Point source
+        isource = N // 2
+        Ydot[isource] += 0.01 / dx
+        return 0
+
+    def jtv(self, vvec, Jvvec, t, yvec, fyvec, tmpvec):
+        N, k, dx = self.N, self.k, self.dx
+        V = N_VGetArrayPointer(vvec)
+        JV = N_VGetArrayPointer(Jvvec)
+        JV[:] = 0.0
+        c1 = k / dx / dx
+        c2 = -2.0 * k / dx / dx
+        # Vectorized tridiagonal product
+        JV[1:-1] = c1 * V[:-2] + c2 * V[1:-1] + c1 * V[2:]
+        JV[0] = 0.0
+        JV[-1] = 0.0
+        return 0
+
+
+def main():
+    # Problem parameters
+    N = 201
+    k = 0.5
+    T0 = 0.0
+    Tf = 1.0
+    Nt = 10
+    reltol = 1e-6
+    abstol = 1e-10
+
+    status, sunctx = SUNContext_Create(SUN_COMM_NULL)
+    y = N_VNew_Serial(N, sunctx)
+
+    problem = Heat1DProblem(N, k)
+    problem.set_init_cond(y)
+
+    # Call ARKStepCreate to initialize the ARK timestepper module and
+    # specify the right-hand side function in y'=f(t,y), the initial time
+    # T0, and the initial dependent variable vector y.  Note: since this
+    # problem is fully implicit, we set f_E to NULL and f_I to f. */
+    ark = ARKStepCreate(
+        None,  # f_E (explicit)
+        lambda t, yvec, ydotvec, _: problem.f(t, yvec, ydotvec),  # f_I (implicit)
+        T0,
+        y,
+        sunctx,
+    )
+
+    # Set routines
+    status = ARKodeSStolerances(ark.get(), reltol, abstol)
+    assert status == ARK_SUCCESS
+
+    status = ARKodeSetMaxNumSteps(ark.get(), 10000)
+    assert status == ARK_SUCCESS
+
+    status = ARKodeSetPredictorMethod(ark.get(), 1)
+    assert status == ARK_SUCCESS
+
+    # PCG linear solver with no preconditioning, with up to N iterations
+    LS = SUNLinSol_PCG(y, 0, N, sunctx)
```

Comment:
We should update the C example to use the enum as well:
```suggestion
    LS = SUNLinSol_PCG(y, SUN_PREC_NONE, N, sunctx)
```